src/focus.service.ts

```diff
@@ -507,19 +507,26 @@ export class FocusService {
     });
   }
 
-  private getFocusableElement(direction: Direction, root: HTMLElement): HTMLElement | null {
-    let el: HTMLElement | null | undefined = this.findNextFocusable(direction, root);
+  private getFocusableElement(direction: Direction, root: HTMLElement, ignore?: HTMLElement): HTMLElement | null {
+    let el: HTMLElement | null | undefined = this.findNextFocusable(direction, root, ignore);
     if (!el) {
       return null;
     }
     const directive = this.registry.find(el);
     if (directive && directive.arcFocusInside) {
-      el = this.getFocusableElement(direction, el);
+      const insideEl = this.getFocusableElement(direction, el);
+
+      // get focusable again if no focusable elements inside the current element
+      if (insideEl === null) {
+        el = this.getFocusableElement(direction, root, el);
+      } else {
+        el = insideEl;
+      }
```

Comment:
how about `return insideEl || this.getFocusableElement(direction, root, el);`

Reply:
That looks nice. Done.